storage/innobase/fts/fts0fts.cc

```diff
@@ -2639,11 +2640,13 @@ fts_cmp_set_sync_doc_id(
 	fts_table.type = FTS_COMMON_TABLE;
 	fts_table.table = table;
 
-	trx = trx_create();
-	if (srv_read_only_mode) {
-		trx_start_internal_read_only(trx);
-	} else {
-		trx_start_internal(trx);
+	if (trx == nullptr) {
+		trx = trx_create();
```

Comment:
I think that the `trx_create()` call must be moved immediately before the `retry:` label. If we reach the line `goto retry`, we would leak a transaction object. We should fix also that bug while we are at this.